src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -43,27 +48,73 @@
     </xsd:complexType>
 
     <xsd:complexType name="ssl-contexts-type">
+        <xsd:annotation>
+            <xsd:documentation>
+                Complex type to contain SSLContext definitions that can subsequently be matched by the ssl context rules.
+            </xsd:documentation>
+        </xsd:annotation>
         <xsd:choice minOccurs="0" maxOccurs="unbounded">
             <xsd:element name="ssl-context" type="ssl-context-type"/>
-            <xsd:element name="default-ssl-context" type="name-type"/>
+            <xsd:element name="default-ssl-context" type="name-type">
+            <xsd:annotation>
+                <xsd:documentation>
+                    The default SSLContext.
+                </xsd:documentation>
+            </xsd:annotation>
+            </xsd:element>
         </xsd:choice>
     </xsd:complexType>
 
     <xsd:complexType name="ssl-context-type">
+        <xsd:annotation>
+            <xsd:documentation>
+                SSLContext definition.
+            </xsd:documentation>
+        </xsd:annotation>
         <xsd:all>
             <xsd:element name="key-store-ssl-certificate" type="key-store-ref-type" minOccurs="0"/>
             <xsd:element name="trust-store" type="trust-store-ref-type" minOccurs="0"/>
             <xsd:element name="trust-manager" type="trust-manager-type" minOccurs="0"/>
-            <xsd:element name="cipher-suite" type="selector-type" minOccurs="0"/>
-            <xsd:element name="protocol" type="names-list-type" minOccurs="0"/>
-            <xsd:element name="provider-name" type="name-type" minOccurs="0"/>
+            <xsd:element name="cipher-suite" type="selector-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Configuration to filter the enabled cipher suites. 
+                        The format of the selector is org.wildfly.security.ssl.CipherSuiteSelector.fromString(selector).
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="protocol" type="names-list-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Define a space separated list of the protocols to be supported.
```

Comment:
"...of SSLContext protocols..."
maybe link to https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#SSLContext for list of available values, or at least "for example TLSv1.2"?